### PR TITLE
Plutus core tidier

### PR DESCRIPTION
**Pull request by kwxm**
The `plutus-core-tidy` PR broke a lot of the cost model stuff because some files contain hard-coded paths which have now changed. I've fixed this and tidied things up a bit (putting the R scripts, CSV files, and JSON files all in a `data` directory).
I'm also wondering if I should be using `data-files` in `plutus-core.cabal` in conjuction with `Paths_plutus_core` to access things in a path-independent way (particularly wrt `costModel.json`, which we need to access at runtime). (Later: yes and no: see SCP-2005).

Pre-submit checklist:
- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
- PR
    - [ ] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested
- If you updated any cabal files or added Haskell packages:
    - [ ] `nix-shell shell.nix --run updateMaterialized` to update the materialized Nix files
    - [ ] Update `hie-*.yaml` files if needed
- If you changed any Haskell files:
    - [ ] `nix-shell shell.nix --run fix-stylish-haskell` to fix any formatting issues
- If you changed any Purescript files:
    - [ ] `nix-shell shell.nix --run fix-purty` to fix any formatting issues

Pre-merge checklist:
- [ ] Someone approved it
- [ ] Commits have useful messages
- [ ] Review clarifications made it into the code
- [ ] History is moderately tidy; or going to squash-merge

### CODE REVIEW

**Comment by michaelpj** [comment]
Oops, thanks for fixing things. I thought I tried running most of them and they seemed to work... evidently not.

File: plutus-core/cost-model/budgeting-bench/Bench.hs [hunk 1]
```diff
@@ -124,10 +141,30 @@ benchTwoInt builtinName =
...
+-- (NOT `plutus`, where `default.nix` is).  See SCP-2005.
 main :: IO ()
 main = do
-    createDirectoryIfMissing True "budgeting-bench/csvs/"
-    defaultMainWith (defaultConfig { C.csvFile = Just $ "budgeting-bench/csvs/benching.csv" }) $ (benchTwoInt <$> twoIntNames) <> (benchTwoByteStrings <$> [Concatenate]) <> (benchBytestringOperations <$> [DropByteString, TakeByteString]) <> (benchHashOperations <$> [SHA2, SHA3]) <> (benchSameTwoByteStrings <$> [EqByteString, LtByteString, GtByteString]) <> [benchVerifySignature] <> benchComparison
-    pure ()
-    where
-        twoIntNames = [AddInteger, SubtractInteger, MultiplyInteger, DivideInteger, QuotientInteger, RemainderInteger, ModInteger, LessThanInteger, LessThanEqInteger, GreaterThanInteger, GreaterThanEqInteger, EqInteger]
+  let dataDir = "cost-model" </> "data" :: FilePath
+      csvFile = dataDir </> "benching4.csv"
```
Comment (michaelpj) [inline plutus-core/cost-model/budgeting-bench/Bench.hs:153]: benching4?

**Review by michaelpj — APPROVED** [review]

File: plutus-core/cost-model/budgeting-bench/Bench.hs (hunk 1 above)
Comment (kwxm) [reply plutus-core/cost-model/budgeting-bench/Bench.hs:153]: Oooh, well-spotted.  That's left over from when I was trying to discover where the different tools were writing files.

Comment (kwxm) [reply plutus-core/cost-model/budgeting-bench/Bench.hs]: ... and we don't need the type annotation on the preceding line either.

**Pull request merged**